include/deal.II/numerics/smoothness_estimator.h

```diff
@@ -0,0 +1,268 @@
+   *    \end{array}\right)
+   *    \left(\begin{array}{c}
+   *    \beta \\ -\mu
+   *    \end{array}\right)
+   *    =
+   *    \left(\begin{array}{c}
+   *    \sum_{{\bf k}, |{\bf k}|\le N} \ln |\hat U_{{\bf k}}|
+   *    \\
+   *    \sum_{{\bf k}, |{\bf k}|\le N} \ln |\hat U_{{\bf k}}| \ln |{\bf k}|
+   *    \end{array}\right)
+   * @f]
+   * Solving for $\beta$ and $\mu$ is nothing else but a linear regression fit
+   * and to do that we will use FESeries::linear_regression().
+   *
+   * While we are not particularly interested in the actual value of
+   * $\beta$, the formula above gives us a mean to calculate the value of
+   * the exponent $\mu$ that we can then use to determine that
+   * $\hat u(\hat{\bf x})$ is in $H^s(\hat K)$ with $s=\mu-\frac d2$. These
+   * Sobolev indices $s$ will suffice as our smoothness estimators and will be
+   * calculated on each cell for any provided solution.
+   *
+   * @note An extensive demonstration of the use of these functions is provided in step-27.
+   *
+   * The @p regression_strategy parameter determines which norm will be used on the subset of
+   * coeffiecients $\mathbf{k}$ with the same absolute value $|\mathbf{k}|$.
+   * Default is VectorTools::Linfty_norm for a maximum approximation.
+   *
+   * Smoothness indicators will be calculated for each solution in @p all_solutions
+   * and stored in @p all_smoothness_indicators in the same order.
+   *
+   * An individual @p fe_series object can be supplied, which has to be constructed with the
+   * same FECollection object as the @p dof_handler.
+   *
+   * @ingroup numerics
+   * @author Denis Davydov, 2016, Marc Fehling, 2018
+   *
+   */
+  template <int dim, int spacedim, typename VectorType>
+  void
+  fourier_coeff_decay(
+    FESeries::Fourier<dim, spacedim> &     fe_series,
+    const hp::DoFHandler<dim, spacedim> &  dof_handler,
+    const std::vector<const VectorType *> &all_solutions,
+    const std::vector<Vector<float> *> &   all_smoothness_indicators,
+    const VectorTools::NormType regression_strategy = VectorTools::Linfty_norm);
+
+  /**
+   * Same as the function above, only for one @p solution vector.
+   */
+  template <int dim, int spacedim, typename VectorType>
+  void
+  fourier_coeff_decay(
+    FESeries::Fourier<dim, spacedim> &   fe_series,
+    const hp::DoFHandler<dim, spacedim> &dof_handler,
+    const VectorType &                   solution,
+    Vector<float> &                      smoothness_indicators,
+    const VectorTools::NormType regression_strategy = VectorTools::Linfty_norm);
+
+  /**
+   * Same as the function above, but with a default configuration for the chosen
+   * series expansion, using 2-point Gaussian quadrature for each finite
+   * element.
+   *
+   * Provide the desired series expansion as a template argument, i.e.
+   * @code
+   *    SmoothnessEstimator::estimate_by_coeff_decay<FESeries::Fourier<dim>>(
+   *      dof_handler, all_solutions, all_smoothness_indicators);
```

Comment:
I don't understand this comment. The function does not have such a template argument, and the function's name is not `estimate_by_coeff_decay` either... ??

Reply:
This has been an important detail in my initial commit, in which I wanted to supply the series expansion as a template argument. Since we refrained from a generalization (for now), this comment makes no sense anymore and will be removed.